From a6c029c905ba969d279c9874aa1324e40e125586 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maciej=20Blizi=C5=84ski?= <maciej.blizinski@gmail.com>
Date: Wed, 3 Sep 2014 22:29:57 +0100
Subject: [PATCH 7/8] Add Makefile.am and configure.ac

Sources used:
http://www.gnu.org/software/automake/manual/automake.html
https://www.flameeyes.eu/autotools-mythbuster/

DEFAULT_FDIR is no longer necessary because --datadir can be used instead, and
.fmt files will be put in the ${pkgdatadir}/abcm2ps directory.
---
 Makefile.am  | 34 ++++++++++++++++++++++++++++++++++
 configure.ac | 60 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 2 files changed, 94 insertions(+)
 create mode 100644 Makefile.am
 create mode 100644 configure.ac

diff --git a/Makefile.am b/Makefile.am
new file mode 100644
index 0000000..b441095
--- /dev/null
+++ b/Makefile.am
@@ -0,0 +1,34 @@
+bin_PROGRAMS = abcm2ps
+abcm2ps_SOURCES = abc2ps.h abcparse.h front.h slre.h \
+									abc2ps.c abcparse.c buffer.c deco.c draw.c format.c front.c \
+									glyph.c music.c parse.c slre.c subs.c svg.c syms.c
+
+abcm2ps_CPPFLAGS = $(PANGO_CFLAGS)
+abcm2ps_LDADD = $(PANGO_LIBS)
+
+abcm2ps_CPPFLAGS += $(CAIRO_CFLAGS)
+abcm2ps_LDADD += $(CAIRO_LIBS)
+
+abcm2ps_CPPFLAGS += $(PANGOCAIRO_CFLAGS)
+abcm2ps_LDADD += $(PANGOCAIRO_LIBS)
+
+abcm2ps_CPPFLAGS += $(PANGOFT2_CFLAGS)
+abcm2ps_LDADD += $(PANGOFT2_LIBS)
+
+abcm2ps_CPPFLAGS += $(GOBJECT_CFLAGS)
+abcm2ps_LDADD += $(GOBJECT_LIBS)
+
+# for SVG
+abcm2ps_LDADD += -lm
+
+dist_doc_DATA = README ChangeLog NEWS *.abc *.eps *.txt
+dist_pkgdata_DATA = *.fmt
+
+# Requires slre
+# bin_PROGRAMS += abcmfe
+# abcmfe_SOURCES = front.c front.h slre.h
+# abcmfe_CPPFLAGS = -DMAIN
+
+TESTS = smoketest.sh
+
+EXTRA_DIST = $(TESTS)
diff --git a/configure.ac b/configure.ac
new file mode 100644
index 0000000..01a787d
--- /dev/null
+++ b/configure.ac
@@ -0,0 +1,60 @@
+dnl Configure source for abcm2ps -*- sh -*-
+dnl Copyright (C) 2001-2014 JF Moine.
+
+AC_PREREQ(2.59)
+AC_INIT([abcm2ps], [8.2.2])
+
+AM_INIT_AUTOMAKE([1.11 silent-rules])
+
+VDATE='August 9, 2014'
+
+dnl Checks for programs.
+AC_PROG_CC
+AC_PROG_INSTALL
+
+AC_CONFIG_HEADER([config.h])
+
+dnl Checks for header files.
+AC_HEADER_STDC
+AC_CHECK_HEADER([malloc.h])
+
+PKG_PROG_PKG_CONFIG
+
+dnl Check for options
+AC_ARG_ENABLE(pango,
+[  --enable-pango          Use pango if available [[default: yes]]],
+[checkpango="$enableval"],
+[checkpango="yes"])
+
+AS_IF([test "x$checkpango" = "xyes"], [
+  PKG_CHECK_MODULES(PANGO, [pango])
+  PKG_CHECK_MODULES(CAIRO, [cairo])
+  PKG_CHECK_MODULES(PANGOCAIRO, [pangocairo])
+  PKG_CHECK_MODULES(GOBJECT, [gobject-2.0])
+  PKG_CHECK_MODULES(PANGOFT2, [pangoft2])
+  # Additional check to see if we can link.
+  AC_SEARCH_LIBS([pango_version], [pango-1.0],
+                 AC_DEFINE([HAVE_PANGO], [1], [Use the Pango library]),
+                 AC_MSG_ERROR([Unable to find the pango-1.0 library.]))
+])
+
+AC_ARG_ENABLE(a4,
+[  --enable-a4             Handle european A4 format instead of default US letter],
+[if test "$enableval" = "yes"; then
+  AC_DEFINE([A4_FORMAT], [], [Handle the A4 format])dnl
+fi])
+
+AC_ARG_ENABLE(deco-is-roll,
+[  --enable-deco-is-roll   Have ~ as roll instead of twiddle],
+[if test "$enableval" = "yes"; then
+  AC_DEFINE([DECO_IS_ROLL], [], [Have ~ as roll instead of twiddle])dnl
+fi])
+
+dnl Build characteristics
+AC_SUBST(CPPPANGO)
+
+AC_DEFINE_UNQUOTED([VDATE], ["$VDATE"], [Date of last modifications])
+AC_DEFINE([DEFAULT_FDIR], ["$pkgdatadir"], [Location of .fmt files])
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
-- 
2.0.3

